docker/main/ngen-partitioner/Dockerfile

```diff
@@ -0,0 +1,48 @@
+ARG DOCKER_INTERNAL_REGISTRY
+FROM ${DOCKER_INTERNAL_REGISTRY}/ngen-deps:latest
+
+#Passing the ARG variables from compose via .env file will squash these defaults with empty strings
+#Seems like the work around is to replicate the default values in the build env, or to check for
+#empty and set to default as is shown commented out below.
+ENV PATH "${WORKDIR}/bin:${PATH}"
+
+ARG REPO_URL=https://github.com/NOAA-OWP/ngen.git
+ARG BRANCH=master
+ARG COMMIT
+
+ENV BOOST_ROOT=${WORKDIR}/boost
+
+ARG BUILD_PARALLEL_JOBS
+ARG BMI_C_ACTIVE=ON
+ARG PYTHON_ACTIVE=ON
+ARG MPI_ACTIVE=ON
+# Note that, at least for now, these are off by default until the functionality is both generally available and needed
+ARG BMI_FORTRAN_ACTIVE=OFF
+ARG ROUTING_ACTIVE=OFF
+ARG PARTITIONER_EXECUTABLE
+
+RUN git clone --single-branch --branch $BRANCH $REPO_URL \
+    && cd ./ngen \
+    && if [ "x$COMMIT" != "x" ]; then git checkout $COMMIT; fi \
+    && git submodule update --init \
```

Comment:
This may be something you're already working on, but this causes the CFE and TOPMODEL repos to be cloned a second time...which is a surprisingly large amount of bandwidth and storage...maybe me running on my LTE connection is the only person that will ever care, but it definitely makes a difference there.